fe/fe-core/src/main/java/com/starrocks/catalog/Column.java

```diff
@@ -713,7 +713,7 @@ public String toSqlWithoutAggregateTypeName(Map<ColumnId, Column> idToColumn) {
                 sb.append("DEFAULT ").append("(").append(defaultExpr.getExpr()).append(") ");
             }
         }
-        if (defaultValue != null && !type.isOnlyMetricType()) {
+        if (defaultExpr == null && defaultValue != null && !type.isOnlyMetricType()) {
```

Comment:
some advise
1. you can refactor this part of logic to make it more readable 
```
if (defalutExpr != null) {
} else {
     if (isAutoIncrement) {
     } else if (defaultValue != null && !type.isOnlyMetricType()) {
     }
}
```
2. should add a test case to cover this bug